### PR TITLE
Formatting Event Text in Email and GOC for better visualisation

### PR DESCRIPTION
**Pull request by studanshu**
Show "Notification:" only if there are more than 1 notification
Show "Triggered by:" only if there are more than 1 trigger
Show "secondaryThreshold" only when condition type is BETWEEN

### CODE REVIEW

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/EmailNotifier.java [hunk 1]
```diff
@@ -119,8 +120,13 @@ protected void sendAdditionalNotification(NotificationContext context) {
     }
 
     private String getEmailSubject(NotificationContext context) {
-        return "[Argus] Notification for Alert: " + context.getAlert().getName() + 
-        		" Notification: "+ context.getNotification().getName() + " Trigger:" + context.getTrigger().getName();
+        String currentSubject = "[Argus] Notification for Alert: " + context.getAlert().getName();
+        Alert currentAlert = context.getAlert();
+        if (currentAlert.getNotifications().size() > 1)
```
Comment (sundeepsf) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/EmailNotifier.java]: should this be >1 or 0?

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/EmailNotifier.java [hunk 2]
```diff
@@ -155,8 +161,11 @@ protected String getEmailBody(NotificationContext context, NotificationStatus no
 
         sb.append(MessageFormat.format("<h3>Alert {0}  was {1} at {2}</h3>", context.getAlert().getName(), notificationMessage,
                 DATE_FORMATTER.get().format(new Date(context.getTriggerFiredTime()))));
-        sb.append(MessageFormat.format("<b>Notification:  </b> {0}<br/>", notification.getName()));
-        sb.append(MessageFormat.format("<b>Triggered by:  </b> {0}<br/>", trigger.getName()));
+        Alert currentAlert = notification.getAlert();
+        if(currentAlert.getNotifications().size() > 1)
```
Comment (sundeepsf) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/EmailNotifier.java]: same here

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/GOCNotifier.java [hunk 3]
```diff
@@ -261,11 +262,13 @@ protected void _sendAdditionalNotification(NotificationContext context, Notifica
 	 */
 	protected String getGOCMessageBody(Notification notification, Trigger trigger, NotificationContext context) {
 		StringBuilder sb = new StringBuilder();
-
+		Alert currentAlert = notification.getAlert();
 		sb.append(MessageFormat.format("Alert {0}  was triggered at {1}\n", context.getAlert().getName(),
 				DATE_FORMATTER.get().format(new Date(context.getTriggerFiredTime()))));
-		sb.append(MessageFormat.format("Notification:  {0}\n", notification.getName()));
-		sb.append(MessageFormat.format("Triggered by:  {0}\n", trigger.getName()));
+		if(currentAlert.getNotifications().size() > 1)
```
Comment (sundeepsf) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/GOCNotifier.java]: same

**Comment by studanshu** [comment]
>1 as per the work item.

https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000005PuiIIAS/view

Regards
Sudhanshu Bahety
Associate Member of Technical Staff, Salesforce com

**Pull request merged**